src/TraceEvent/Computers/StartStopActivityComputer.cs

```diff
@@ -920,8 +920,23 @@ void FixAndProcessAdoNetEvents(TraceEvent data)
                         return;
                     StartStopActivity creator = GetCurrentStartStopActivity(thread, data);
 
-                    Debug.Assert(data.payloadNames[1] == "dataSource");
-                    var extraStartInfo = data.PayloadValue(1) as string;
+                    string extraStartInfo = null;
+                    if (3 < data.payloadNames.Length)
+                    {
+                        Debug.Assert(data.payloadNames[1] == "dataSource");
+                        Debug.Assert(data.payloadNames[3] == "commandText");
+                        string dataSource = data.PayloadValue(1).ToString();
+                        string commandText = data.PayloadValue(3).ToString();
+                        extraStartInfo = "DS=" + dataSource;
+                        if (!string.IsNullOrEmpty(commandText))
+                        {
+                            // Don't make it too long
```

Comment:
💡 To me, the truncation seems like a normal coding pattern for "don't make it too long". More interesting to me is the value of `extraStartInfo`. I would guess that this allows people to identify specific queries getting executed from within stack trace views, but this may or may not be correct. If there is a concise explanation for the net value of this block it might be a good replacement for this comment.